src/Estimators/src/LeggedOdometry.cpp

```diff
@@ -0,0 +1,907 @@
+                                                       FloatingBaseEstimator::ModelComputations& modelComp,
+                                                       std::map<int, EstimatedContact>& contactStates)
+{
+    for (auto& obs : meas.stampedContactsStatus)
+    {
+        auto& measContact = obs.second;
+        auto& idx = obs.first;
+
+        if (contactStates.find(idx) != contactStates.end())
+        {
+            contactStates.at(idx).setContactStateStamped({measContact.isActive, measContact.switchTime});
+            contactStates.at(idx).lastUpdateTime = measContact.lastUpdateTime;
+        }
+        else
+        {
+            BipedalLocomotion::Contacts::EstimatedContact newContact;
+            newContact.setContactStateStamped({measContact.isActive, measContact.switchTime});
+            newContact.lastUpdateTime = measContact.lastUpdateTime;
+            newContact.index = idx;
+            newContact.name = modelComp.kinDyn()->getFrameName(idx);
+            contactStates[idx] = newContact;
+        }
+    }
+    meas.stampedContactsStatus.clear();
+}
+
+iDynTree::FrameIndex LeggedOdometry::Impl::getLatestSwitchedContact(const std::map<int, BipedalLocomotion::Contacts::EstimatedContact>& contacts)
+{
+    std::string_view printPrefix = "[LeggedOdometry::Impl::getLatestContact] ";
+    if (contacts.size() < 1)
+    {
+        std::cerr << printPrefix << "No contact data available." << std::endl;
+        return iDynTree::FRAME_INVALID_INDEX;
+    }
+
+    bool atleastOneActiveContact{false};
+    int latestContactIdx{-1};
+    double latestTime{-1.0}; // assuming time cannot be negative
+    for (auto& [idx, contact] : contacts)
+    {
+        if (contact.isActive)
+        {
+            atleastOneActiveContact = true;
+            if (contact.switchTime > latestTime)
+            {
+                latestContactIdx = idx;
+                latestTime = contact.switchTime;
+            }
+        }
+    }
+
+    if (!atleastOneActiveContact)
+    {
+        std::cerr << printPrefix << "No active contacts." << std::endl;
+        return iDynTree::FRAME_INVALID_INDEX;
+    }
+    return latestContactIdx;
+}
+
+iDynTree::FrameIndex LeggedOdometry::Impl::getLastActiveContact(const std::map<int, BipedalLocomotion::Contacts::EstimatedContact>& contacts, const int& activeIndex)
+{
+    std::string_view printPrefix = "[LeggedOdometry::Impl::getLastActiveContact] ";
+    if (contacts.size() < 1)
+    {
+        std::cerr << printPrefix << "No contact data available." << std::endl;
+        return iDynTree::FRAME_INVALID_INDEX;
+    }
+
+    bool atleastOneActiveContact{false};
+    if (contacts.find(activeIndex) != contacts.end())
+    {
+        if (contacts.at(activeIndex).isActive)
+        {
+            return activeIndex;
+        }
+    }
+
+    double latestTime{std::numeric_limits<double>::max()}; // assuming time cannot be negative
+    int latestContactIdx{-1};
+    for (auto& [idx, contact] : contacts)
```

Comment:
`const` also here

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/151/commits/914786f7529f478c66665697e516685f9953078b

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/151/commits/5a8f5db8a89c629246418880188e4dabf816e067